cmd/talosctl/cmd/mgmt/cluster/create.go

```diff
@@ -646,12 +684,13 @@ func create(ctx context.Context, flags *pflag.FlagSet) error {
 			}
 		}
 
+		endpointList = []string{fmt.Sprintf("%s:%d", forceEndpoint, apiPort)}
```

Comment:
this is certainly a bug, this should be merged via the `case defaultEndpoint != ""` branch on line 667

the Docker provisioner should return "127.0.0.1:dynamic_port" as the `defaultEndpoint`